### PR TITLE
New Resource `azurerm_machine_learning_workspace_hub`

### PR DESCRIPTION
**Pull request by xuzhang3**
Add support  for new Machine learning workspace type `Hub` - New resource `azurerm_machine_learning_workspace_hub`
API: https://learn.microsoft.com/en-us/rest/api/azureml/workspaces/create-or-update?view=rest-azureml-2024-04-01&tabs=HTTP

```log
=== RUN   TestAccMachineLearningWorkspaceHub_basic
=== PAUSE TestAccMachineLearningWorkspaceHub_basic
=== RUN   TestAccMachineLearningWorkspaceHub_requiresImport
=== PAUSE TestAccMachineLearningWorkspaceHub_requiresImport
=== RUN   TestAccMachineLearningWorkspaceHub_complete
=== PAUSE TestAccMachineLearningWorkspaceHub_complete
=== RUN   TestAccMachineLearningWorkspaceHub_completeUpdate
=== PAUSE TestAccMachineLearningWorkspaceHub_completeUpdate
=== RUN   TestAccMachineLearningWorkspaceHub_userAssignedIdentity
=== PAUSE TestAccMachineLearningWorkspaceHub_userAssignedIdentity
=== RUN   TestAccMachineLearningWorkspaceHub_systemAssignedUserAssignedIdentity
=== PAUSE TestAccMachineLearningWorkspaceHub_systemAssignedUserAssignedIdentity
=== RUN   TestAccMachineLearningWorkspaceHub_identityUpdate
=== PAUSE TestAccMachineLearningWorkspaceHub_identityUpdate
=== RUN   TestAccMachineLearningWorkspaceHub_identityTypeUpdate
=== PAUSE TestAccMachineLearningWorkspaceHub_identityTypeUpdate
=== RUN   TestAccMachineLearningWorkspaceHub_userAssignedAndCustomManagedKey
=== PAUSE TestAccMachineLearningWorkspaceHub_userAssignedAndCustomManagedKey
=== RUN   TestAccMachineLearningWorkspaceHub_purgeSoftDelete
=== PAUSE TestAccMachineLearningWorkspaceHub_purgeSoftDelete
=== CONT  TestAccMachineLearningWorkspaceHub_basic
=== CONT  TestAccMachineLearningWorkspaceHub_systemAssignedUserAssignedIdentity
=== CONT  TestAccMachineLearningWorkspaceHub_userAssignedAndCustomManagedKey
=== CONT  TestAccMachineLearningWorkspaceHub_identityTypeUpdate
=== CONT  TestAccMachineLearningWorkspaceHub_completeUpdate
=== CONT  TestAccMachineLearningWorkspaceHub_complete
=== CONT  TestAccMachineLearningWorkspaceHub_requiresImport
=== CONT  TestAccMachineLearningWorkspaceHub_identityUpdate
=== CONT  TestAccMachineLearningWorkspaceHub_userAssignedIdentity
--- PASS: TestAccMachineLearningWorkspaceHub_userAssignedIdentity (503.16s)
--- PASS: TestAccMachineLearningWorkspaceHub_purgeSoftDelete (503.82s)
--- PASS: TestAccMachineLearningWorkspaceHub_systemAssignedUserAssignedIdentity (519.97s)
--- PASS: TestAccMachineLearningWorkspaceHub_identityUpdate (527.25s)
--- PASS: TestAccMachineLearningWorkspaceHub_basic (806.15s)
--- PASS: TestAccMachineLearningWorkspaceHub_identityTypeUpdate (979.41s)
--- PASS: TestAccMachineLearningWorkspaceHub_requiresImport (1030.74s)
--- PASS: TestAccMachineLearningWorkspaceHub_complete (1836.38s)
--- PASS: TestAccMachineLearningWorkspaceHub_userAssignedAndCustomManagedKey (2106.29s)
--- PASS: TestAccMachineLearningWorkspaceHub_completeUpdate (2348.30s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning       2364.159s
```

## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/machinelearning/machine_learning_workspace_hub_resource.go [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+			Optional: true,
+		},
+
+		"key_vault_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateKeyVaultID,
+			// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+			DiffSuppressFunc: suppress.CaseDifference,
```
Comment (katbyte) [inline internal/services/machinelearning/machine_learning_workspace_hub_resource.go:93]: can you please reach out to the service team about this? the linked issue is 4 years old and there is no indication of any progress on the service team fixing it. could we get an update here?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/machinelearning/machine_learning_workspace_hub_resource.go (hunk 1 above)
Comment (xuzhang3) [reply internal/services/machinelearning/machine_learning_workspace_hub_resource.go:93]: Will check with the service team, the issue appears to been fxied.

File: internal/services/machinelearning/machine_learning_workspace_hub_resource.go [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+					"key_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+					},
+					"user_assigned_identity_id": {
+						Type:             pluginsdk.TypeString,
+						Optional:         true,
+						ValidateFunc:     commonids.ValidateUserAssignedIdentityID,
+						DiffSuppressFunc: suppress.CaseDifference,
```
Comment (katbyte) [inline internal/services/machinelearning/machine_learning_workspace_hub_resource.go:134]: how come this is required? if it is that is likely a bug in the API - could you open an issue on the rest epcs and link it here?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/machinelearning/machine_learning_workspace_hub_resource.go (hunk 2 above)
Comment (xuzhang3) [reply internal/services/machinelearning/machine_learning_workspace_hub_resource.go:134]: will check this with service team. As far as I know this issue has not been fixed and we need to suppress case insensitive diff

**Comment by stephybun** [comment]
Thanks @xuzhang3. These resources should reflect the commercial name seen in the documentation as well as the Portal and probably warrants the creation of a new service package `aiservices`. I've opened #27424 to add the Hub and Project resources together under this new service package. I'm going to go ahead and close this since it's been superseded by #27424.

**Pull request closed without merging**